prometheus-exporter/src/main/java/be/xplore/githubmetrics/prometheusexporter/workflowrun/WorkflowRunBuildTimesOfLastDayExporter.java

```diff
@@ -0,0 +1,146 @@
+
+        this.publishWorkflowRunsBuildTimesGauges(buildTimesMap);
+    }
+
+    private void publishWorkflowRunsBuildTimesGauges(
+            Map<String, Map<WorkflowRunStatus, Integer>> buildTimesMap
+    ) {
+        for (Map.Entry<String, Map<WorkflowRunStatus, Integer>> metric : buildTimesMap.entrySet()) {
+            for (Map.Entry<WorkflowRunStatus, Integer> entry : metric.getValue().entrySet()) {
+                Gauge.builder(metric.getKey(),
+                                entry,
+                                Map.Entry::getValue
+                        ).tag("status", entry.getKey().toString())
+                        .strongReference(true)
+                        .register(this.registry);
+
+            }
+
+        }
+    }
+
+    private Map<WorkflowRunStatus, Integer> getAverageBuildTimes(
+            List<WorkflowRun> workflowRuns
+    ) {
+        Map<WorkflowRunStatus, Integer> workflowRunsExportMap =
+                this.createExportMap();
+        for (var workflowRunsOfStatus : this.getWorkflowRunsByStatus(workflowRuns).entrySet()) {
+            workflowRunsExportMap.put(
+                    workflowRunsOfStatus.getKey(),
+                    (int) round(workflowRunsOfStatus.getValue()
+                            .stream()
+                            .mapToDouble(WorkflowRun::getBuildTime)
+                            .average()
+                            .orElse(0.0)));
+        }
+        return workflowRunsExportMap;
+    }
+
+    private Map<WorkflowRunStatus, List<WorkflowRun>> getWorkflowRunsByStatus(
+            List<WorkflowRun> workflowRuns
+    ) {
+        Map<WorkflowRunStatus, List<WorkflowRun>> workflowRunByStatusMap =
+                this.getEmptyWorkflowRunsByStatus();
+
+        workflowRuns.forEach(workflowRun ->
+                workflowRunByStatusMap.get(workflowRun.getStatus()).add(workflowRun)
+        );
+
+        return workflowRunByStatusMap;
+    }
+
+    private Map<WorkflowRunStatus, List<WorkflowRun>> getEmptyWorkflowRunsByStatus() {
+        Map<WorkflowRunStatus, List<WorkflowRun>> workflowRunsByStatusMap
+                = new EnumMap<>(WorkflowRunStatus.class);
+
+        Stream.of(WorkflowRunStatus.values()).forEach(
+                runStatus -> workflowRunsByStatusMap.put(runStatus, new ArrayList<>()));
+
+        return workflowRunsByStatusMap;
+    }
+
+    private Map<WorkflowRunStatus, Integer> getTotalBuildTimes(
+            List<WorkflowRun> workflowRuns
+    ) {
+        Map<WorkflowRunStatus, Integer> workflowRunExportMap = createExportMap();
+        workflowRuns.forEach(workflowRun ->
+                workflowRunExportMap.put(
+                        workflowRun.getStatus(),
+                        workflowRun.getBuildTime() +
+                                workflowRunExportMap.get(workflowRun.getStatus())
+                ));
+        return workflowRunExportMap;
```

Comment:
Sorry to do this to you only now but cant this use some of the code you wrote up there? What I mean is that if you have a list of doubles then you can both get the average and the total from that same list no?